compiler/rustc_typeck/src/check/demand.rs

```diff
@@ -140,6 +140,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             Err(e) => e,
         };
 
+        self.set_tainted_by_errors();
```

Comment:
Ha! Smart!

Reply:
Haha thanks, super useful someone had already put that in!